CMakeLists.txt

```diff
@@ -127,6 +127,17 @@ option(LBANN_WITH_NVPROF
 option(LBANN_WITH_PYTHON
   "Install Python frontend and enable embedded Python" ON)
 
+option(LBANN_WITH_PYTHON_CXX
```

Comment:
```suggestion
option(LBANN_WITH_PYTHON_LIB
```
Pedantic suggestion. It's technically linking to a C library, so `LBANN_WITH_PYTHON_C` might be another alternative.

Reply:
Oh, I was thinking "Build LBANN with Python embedded in the (LBANN) C++ code." Don't feel strongly.

Reply:
How about `LBANN_WITH_EMBEDDED_PYTHON`?

Reply:
I changed it to LBANN_WITH_EMBEDDED_PYTHON and also updated the spack build front end to expose these flags.